crates/crane/src/typer.rs

```diff
@@ -175,16 +174,6 @@ impl Typer {
             }
         }
 
-        let mut called_fns_collector = CalledFnsCollector::new();
-
-        for item in &module.items {
-            called_fns_collector.visit_item(item);
-        }
-
-        for called_fn in called_fns_collector.called_fns {
-            self.ensure_function_exists(&called_fn)?;
```

Comment:
While debugging something related to paths I realized that this call to `ensure_function_exists` seemed to be redundant.

We already have a call in `self.infer_function` that seems to handle the cases.